apps/cowswap-frontend/src/modules/trade/hooks/setupTradeState/useSetupTradeState.ts

```diff
@@ -124,7 +125,9 @@ export function useSetupTradeState(): void {
     const tokensAreEmpty = !inputCurrencyId && !outputCurrencyId
 
     const sameTokens =
-      (inputCurrencyId || outputCurrencyId) && inputCurrencyId?.toLowerCase() === outputCurrencyId?.toLowerCase()
+      inputCurrencyId !== EMPTY_TOKEN_ID &&
+      (inputCurrencyId || outputCurrencyId) &&
+      inputCurrencyId?.toLowerCase() === outputCurrencyId?.toLowerCase()
```

Comment:
This is the main fix. Otherwise the app never clears out the inner state.
Widget consumers (Den) want to be able to clear out both sell & buy tokens, which sets their values to `_`.